src/Hyperion/Main.hs

```diff
@@ -141,11 +146,28 @@ doRun bks = do
       throwIO $ DuplicateNames [ n | n:_:_ <- group (sort nms) ]
     foldMap (runBenchmark (sample 100 (fixed 5))) bks
 
-doAnalyze :: Config -> [Benchmark] -> IO ()
-doAnalyze Config{..} bks = do
+doAnalyze
+  :: Config -- ^ Hyperion config.
+  -> Text -- ^ Package name.
+  -> [Benchmark] -- ^ Benchmarks to be run.
+  -> IO ()
+doAnalyze Config{..} packageName bks = do
+    executableName <- getProgName -- Name of the execuatble that launched the benches.
     h <- case configOutputPath of
       Nothing -> return IO.stdout
-      Just path -> IO.openFile path IO.WriteMode
+      Just path ->
+        if path == nullOutputPath
+          then IO.openFile nullOutputPath IO.WriteMode
```

Comment:
You shouldn't need to special case `nullOutputPath`. Notice that the action is the same is in the `else` branch of the second if below.